terms/confusionmatrix_df.tex

```diff
@@ -0,0 +1,199 @@
+In the \textit{1-vs-rest} approach, 2-class CMs are constructed for each class and the remaining classes combined together as shown in Table \ref{tab:cm_1vsN}.
+
+\begin{table}[ht]
+  \centering
+  \fontfamily{ppl}\selectfont
+  \begin{tabular}{ll}
+    \toprule
+    +ve       & -ve \\
+    \midrule
+    M\&M's    & Skittles $\cup$ Smarties \\
+    Skittles  & M\&M's $\cup$ Smarties \\
+    Smarties  & Skittles $\cup$ M\&M's \\
+    \bottomrule
+  \end{tabular}
+  \caption{2-class CMs derived through decomposition of the 3-class CM from Table \ref{tab:cm_sweets} using the 1-vs-rest approach.}
+  \label{tab:cm_1vsN}
+\end{table}
+
+Taking into account the above, CMs are suitable for visualising, evaluating, and comparing the performance of binary or multi-class classifiers. They should be used in conjunction with metrics such as the F-measure to avoid bias, especially if the dataset is unbalanced. For further details on the theoretical aspects of CMs and for practical examples in R refer to \citep{cichosz2014data}; for examples in Python refer to \citep{muller2016introduction}.
+
+The following example is motivated by the samples in the \textit{Scikit-Learn} documentation and the work of  \citep{geron2017hands}. The models in Fig. \ref{fig:cm_model} were trained on 70\% of the \textit{wines} dataset included with Scikit-Learn, whilst the remaining 30\% was kept as a test set to evaluate the model. The dataset has 3 classes and 13 features, but only the results for class 1 and 2; and the alcohol and malic acid features; will be presented. It was scaled by standardising the features using $\frac{x_i - \mu}{\sigma}$, for all $x_i \in S$ using mean $\mu$ and standard deviation $\sigma$.
+
+\begin{figure}
+  \includegraphics{confusion_matrix/model.png}
+  \caption{Decision boundary learned by a linear and non-linear binary classifier.}
+  \label{fig:cm_model}
+\end{figure}
+
+\begin{margintable}
+  \begin{tabular}{lll}
+    \toprule
+                 & Linear & Non-Linear \\
+    \midrule
+    Accuracy     & 0.72   & 0.78 \\
+    Specificity  & 0.77   & 0.77 \\
+    Sensitivity  & 0.70   & 0.78 \\
+    Precision    & 0.84   & 0.86 \\
+    F-score      & 0.76   & 0.82 \\
+    \bottomrule
+  \end{tabular}
+  \caption{Statistics derived from the CMs in Fig. \ref{fig:cm_wines}.}
+  \label{tab:cm_metrics}
+\end{margintable}
+
+As it can be intuitively deduced from Fig. \ref{fig:cm_model}, the decision boundary of the non-linear model is a better fit than the linear model. The CMs in Fig. \ref{fig:cm_wines} also show that non-linear model performs better with a higher TP, and consequently lower TN. The biggest advantage of the non-linear model is the higher Sensitivity resulting in a better F-score.
+
+\begin{figure}
+  \includegraphics{confusion_matrix/cm.png}
+  \caption{The linear classifier has 16 TP, 10 TN, 7 FN and 3 FP, whilst the non-linear classifier has 18 TP, 10 TN, 5 FN and 3 FP.}
+  \label{fig:cm_wines}
+\end{figure}
+
+\index{class options|)}
```

Comment:
index class options ???? no .... this should be a closing \index{confusion matrix|)} but you need an opening tag.

No indexing information throughout ... please add.